### PR TITLE
Update the shims

### PR DESCRIPTION
**Pull request by Elv13**
Extracted from the other pull requests (both the popup, dynamic layout and keygrabber ones)

It try to unify all the coding style of the different squashed commits and have a (fake) commit log.

All the issues that I agreed from @psychon comments in all related PRs are addressed or voluntarily ignored.

The goal of the shims is to be able to load awful and wibox and have semi accurate documentation examples. The core principle of what is and what is not in the shims are:

 * Only add what's being used by the tests, everything else is dead-code
 * Keep the implementation as simple as possible
 * Do not handle errors that would create unwanted side effects, if they are triggered, then the documentation examples are badly written.
 * Ignore corner cases that are not part of the documentation examples, it adds complexity and is dead-code.

### CODE REVIEW

File: tests/examples/shims/awesome.lua [hunk 1]
```diff
@@ -43,9 +43,26 @@ awesome._shim_fake_class = _shim_fake_class
...
 function awesome.register_xproperty()
 end
 
+function awesome.xkb_get_group_names()
+    return "pc+us+inet(evdev)"
+end
+
+function awesome.xkb_get_layout_group()
+    return 0
+end
```
Comment (psychon) [inline tests/examples/shims/awesome.lua]: > All the issues that I agreed from @psychon comments in all related PRs are addressed or voluntarily ignored.

Uhm, I'm not quite sure what "voluntarily ignored" means, so I will not repeat my comments on this code...
(Duplication here, `s.outputs` is still wrong, I stopped reading at this point)

**Comment by Elv13** [comment]
> (Duplication here, s.outputs is still wrong, I stopped reading at this point)

I fixed that one, it was an oversaw. For the other ones, I commented on them in the other PR and it is why they are not addressed. If you disagree with those comments, say so there then I will do something about it. And to repeat myself (put intended), the goal of the shims is to allow `awful` and `wibox` to load+draw in the documentation examples. The shims are not intended to be a full CAPI clone just as the busted/spec ones are equally incomplete. 

The `screen.outputs` "issue" is a non issue because it is only used in `awful/screen.lua:584` as `for _, o in pairs(s.outputs) do`. This the key is unused deadcode and should not be added to the shims because of that. The shims should not add dead code for the sake of completeness (assuming the related unmerged PRs count as "incoming alive code"). It makes them more complex for all the wrong reasons.

**Comment by Elv13** [comment]
Updated without the keygrabber miss handlers. Some issues have comment in the other PR that are awaiting replies (or simply left as-is). Otherwise all other changes should have been implemented unless I missed them.

File: tests/examples/shims/beautiful.lua [hunk 2]
```diff
@@ -29,10 +29,13 @@ end
 local module = {
     fg_normal    = "#000000"  ,
     bg_normal    = "#6181FF7D",
+    bg_focus     = "#AA00FF7D",
     bg_highlight = "#AA00FF7D",
     border_color = "#6181FF"  ,
     border_width = 1.5        ,
 
+    prompt_bg_cursor = "#00FF7D",
```
[the remark this answers is not in the document]
Comment (Elv13) [reply tests/examples/shims/beautiful.lua]: right

**Comment by Elv13** [comment]
Updated again. Note that I simplified `capi.keygrabber to avoid the _private now that it was totally unused

**Comment by Elv13** [comment]
Pushed again, removing the merge issue that caused the xkb function to be there twice was in the wrong commit

**Comment by Elv13** [comment]
Thanks, I will merge this now to be able to get the other PRs in reviewable state. Even if issues are found, it can be fixed later given it's not a user facing codepath (beside the doc, obviously)
